posthog/models/filters/filter.py

```diff
@@ -119,9 +124,11 @@ def __init__(
             raise ValueError("You need to define either a data dict or a request")
 
         self._data = data
-
         self.kwargs = kwargs
-
-        if "team" in kwargs and not self.is_simplified:
-            simplified_filter = self.simplify(kwargs["team"])
-            self._data = simplified_filter._data
+        self.hogql_context = hogql_context or HogQLContext()
```

Comment:
Would it make sense move this to a mixin and make it a @cached_property instead?

Reply:
Sure, let's try.